src/transactions/OperationFrame.cpp

```diff
@@ -127,7 +127,8 @@ bool OperationFrame::isVersionSupported(uint32_t) const
 
 bool
 OperationFrame::checkSignature(SignatureChecker& signatureChecker,
-                               AbstractLedgerTxn& ltx, bool forApply)
+                               AbstractLedgerTxn& ltx, bool forApply,
```

Comment:
I would prefer a solution that doesn't require changes to `OperationFrame::checkSignature`. Any changes here are just working around existing issues in the design of `TransactionFrame` and `OperationFrame`, thereby increasing rather than decreasing complexity.